### DISCUSSION TITLE
Expose auth_header value in BasicAuth as a property

### DISCUSSION THREAD

**Discussion by skewty** [encode/httpx#2654] [Ideas]
I am working with some devices in the wild that oddly require BasicAuth as well as Digest Auth.

It needs the BasicAuth._auth_header value injected into the header prior to the request which itself requires Digest Auth.

I propose making the `auth_header` value accessible by making it property.

```python
class NewBasicAuth(BasicAuth):
    @property
    def auth_header(self) -> str:
        return self._auth_header
```

**Comment by skewty** [comment]
With the above change I can avoid accessing a protected value, avoid doing the bas64 encoding myself (keep it DRY) and or avoid having to subclass upstream BasicAuth class with a subclass that exposes the value.

```python
value_to_be_used_in_header = BasicAuth("someuser", "somepassword").auth_header
```

**Reply by lovelydinosaur** [reply]
Hello.

If you've a custom use case, create and maintain a custom auth class.

**Reply by skewty** [reply]
> Hello.
> 
> If you've a custom use case, create and maintain a custom auth class.

Fair point. But BasicAuth is essentially a "value" class without it's value exposed.
